joomla/joomla-cms
# Fix for Finder (Smart Search) indexer removing spaces in long strings

### Summary of Changes

The indexer parses strings through different string parsers (like `Joomla\Component\Finder\Administrator\Indexer\Parser\Html`).
The main `parse` method (`Joomla\Component\Finder\Administrator\Indexer\Parser::parse()`) attempts to chop long strings in chunks, and then passes each chunk to the `process` method.
It will truncate chunks to the last space found in the string.

But it doesn't include this space in the string. This will cause the spaces to get lost in the string concatenation for the final `$return` string.

This PR fixes the issue.

### Testing Instructions

To test, create an article with this content:

```
<p>aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX aaa XXX</p>
```

and make sure it's indexed by Smart Search.

### Actual result BEFORE applying this Pull Request

Before the PR is applied, using Smart Search you can search for `aaaXXX` (so no space) and get a result.

### Expected result AFTER applying this Pull Request

After the PR is applied (and re-saving the article), you won't find any more results for `aaaXXX` (with no space), but will have results when searching for `aaa XXX`.
